tracer/src/Datadog.Trace/Ci/IntelligentTestRunnerClient.cs

```diff
@@ -71,6 +72,21 @@ public IntelligentTestRunnerClient(string workingDirectory, CIVisibilitySettings
         _workingDirectory = workingDirectory;
         _environment = TraceUtil.NormalizeTag(_settings.TracerSettings.Environment ?? string.Empty) ?? string.Empty;
         _serviceName = NormalizerTraceProcessor.NormalizeService(_settings.TracerSettings.ServiceName) ?? string.Empty;
+        _customConfigurations = null;
+
+        // Extract custom tests configurations from DD_TAGS
+        var globalTags = _settings.TracerSettings.GlobalTags;
+        foreach (var tag in globalTags)
+        {
+            const string testConfigKey = "test.configuration.";
+            if (tag.Key.StartsWith(testConfigKey, StringComparison.OrdinalIgnoreCase))
+            {
+                var key = tag.Key.Substring(testConfigKey.Length);
+                _customConfigurations ??= new Dictionary<string, string>();
+                _customConfigurations[key] = tag.Value;
+            }
+        }
+
```

Comment:
I don't see any tests for this - do you want to add some unit tests at least to confirm the tags are parsed correctly? e.g. would be good to see some example data for what the tags will actually look like, to make sure nothing goes wrong in the parsing of the GlobalTags etc

Reply:
Sure, I'll do a refactor and create a tests for this parser.

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/3417/commits/b39e038b9b573ba10e82992d6b6c0617c52b0e23